adapter/serviceControl/serviceController.go

```diff
@@ -0,0 +1,218 @@
+	var vs []*sc.MetricValueSet
+	for _, v := range values {
+		// TODO Only support request count.
+		if v.Definition.Name != "request_count" {
+			a.logger.Warningf("service control metric got unsupported metric name: %s\n", v.Definition.Name)
+			continue
+		}
+		var mv sc.MetricValue
+		mv.Labels, _ = translate(v.Labels)
+		mv.StartTime = v.StartTime.Format(time.RFC3339Nano)
+		mv.EndTime = v.EndTime.Format(time.RFC3339Nano)
+		i, _ := v.Int64()
+		mv.Int64Value = &i
+
+		ms := &sc.MetricValueSet{
+			MetricName:   "serviceruntime.googleapis.com/api/producer/request_count",
+			MetricValues: []*sc.MetricValue{&mv},
+		}
+		vs = append(vs, ms)
+	}
+
+	ot := now.Format(time.RFC3339Nano)
+	op := &sc.Operation{
+		OperationId:     operationID,
+		OperationName:   "reportMetrics",
+		StartTime:       ot,
+		EndTime:         ot,
+		MetricValueSets: vs,
+		Labels: map[string]string{
+			"cloud.googleapis.com/location": "global",
+		},
+	}
+	return &sc.ReportRequest{
+		Operations: []*sc.Operation{op},
+	}
+}
+
+// translate mixer label key to GCP label, and find the api method. TODO Need general redesign.
+// Current implementation is just adding a prefix / which does not work for some labels.
+// Need to support api_method
+// "serviceruntime.googleapis.com/api_method": method,
+func translate(labels map[string]interface{}) (map[string]string, string) {
+	ml := make(map[string]string)
+	var method string
+	for k, v := range labels {
+		nk := "/" + k
+		ml[nk] = fmt.Sprintf("%v", v)
+		if k == "request.method" {
+			method = ml[nk]
+		}
+	}
+	return ml, method
+}
+
+//TODO Add supports to struct payload.
+func (a *aspect) log(entries []adapter.LogEntry, now time.Time, operationID string) *sc.ReportRequest {
+	var ls []*sc.LogEntry
+	for _, e := range entries {
+		l := &sc.LogEntry{
+			Name:        e.LogName,
+			Severity:    e.Severity.String(),
+			TextPayload: e.TextPayload,
+			Timestamp:   e.Timestamp,
+		}
+		ls = append(ls, l)
+	}
+
+	ot := now.Format(time.RFC3339Nano)
+	op := &sc.Operation{
+		OperationId:   operationID,
+		OperationName: "reportLogs",
+		StartTime:     ot,
+		EndTime:       ot,
+		LogEntries:    ls,
+		Labels:        map[string]string{"cloud.googleapis.com/location": "global"},
+	}
+
+	return &sc.ReportRequest{
+		Operations: []*sc.Operation{op},
+	}
+}
+
+func (a *aspect) Log(entries []adapter.LogEntry) error {
+	b := bytes.NewBufferString("mixer-metric-log-id-")
+	_, err := b.WriteString(uuid.New())
+	if err != nil {
+		return err
+	}
+	rq := a.log(entries, time.Now(), b.String())
+	rp, err := a.service.Services.Report(a.serviceName, rq).Do()
+	if err != nil {
+		return err
+	}
+
+	if a.logger.VerbosityLevel(4) {
+		a.logger.Infof("service control log report for operation id %s\nlogs %v\nresponse %v",
+			rq.Operations[0].OperationId, entries, rp)
+	}
+	return nil
+}
+
+func (a *aspect) LogAccess(entries []adapter.LogEntry) error {
```

Comment:
I'm not sure I follow: if we aren't handling `LogAccess()`, why register as an access logger?  Otherwise, it feels like we should just have an internal `log([]adapter.LogEntry)` method that is called by both `Log()` and `LogAccess()`